### ISSUE TITLE
Output lacks bypass blocks (fails WCAG 2.4.1)

### ISSUE THREAD

**Issue by Azuryllis** [rust-lang/mdBook#2107]
### Problem

mdBook currently does not have *bypass blocks* to skip past content repeated on multiple pages (e.g. the left sidebar and header) when using keyboard navigation. This fails [WCAG 2.4.1](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-skip.html). Definition below is taken from WCAG 2.4.1.
> A [mechanism](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-skip.html#mechanismdef) is available to bypass blocks of content that are repeated on multiple [Web pages](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-skip.html#webpagedef). (Level A)

### Steps

1. Go to an mdbook site (was using https://doc.rust-lang.org/stable/book/)
2. Press 'tab'
3. No bypass block is focused on

### Possible Solution(s)

Both [Astro Starlight](https://starlight.astro.build/) and [Docusaurus](https://docusaurus.io/) have bypass blocks which can be used as examples. (If you go to either of those pages and press 'tab', a "Skip to main content" button appears).

### Version

```text
Was using https://doc.rust-lang.org/stable/book/ to learn and noticed this, so whatever version of mdBook that was made in (also applies to https://rust-lang.github.io/mdBook/ too)
```
